tests/qnn/test_keras.py

```diff
@@ -180,6 +268,53 @@ def c_dummy(inputs, w1, w2, w3, w4, w5, w6, w7, w8=None):
         ):
             KerasLayer(c_dummy, {**w, **{"w8": 1}}, output_dim)
 
+    @pytest.mark.parametrize("n_qubits, output_dim", indices_up_to(1))
+    def test_non_input_defaults_tape_mode(self):
+        """Test that everything works when default arguments that are not the input argument are
+        present in the QNode in tape mode"""
+        if not qml.tape_mode_active():
+            pytest.skip("This functionality is only supported in tape mode.")
+
+        n_qubits = 2
+        output_dim = 2
+
+        dev = qml.device("default.qubit", wires=n_qubits)
+        w = {
+            "w1": (3, n_qubits, 3),
+            "w2": (1,),
+            "w3": 1,
+            "w4": [3],
+            "w5": (2, n_qubits, 3),
+            "w6": 3,
+            "w7": 0,
+        }
+
+        @qml.qnode(dev, interface="tf")
+        def c(inputs, w1, w2, w4, w5, w6, w7, w3=0.5):
+            """A circuit that embeds data using the AngleEmbedding and then performs a variety of
+            operations. The output is a PauliZ measurement on the first output_dim qubits. One set of
+            parameters, w5, are specified as non-trainable."""
+            qml.templates.AngleEmbedding(inputs, wires=list(range(n_qubits)))
+            qml.templates.StronglyEntanglingLayers(w1, wires=list(range(n_qubits)))
+            qml.RX(w2[0], wires=0 % n_qubits)
+            qml.RX(w3, wires=1 % n_qubits)
+            qml.Rot(*w4, wires=2 % n_qubits)
+            qml.templates.StronglyEntanglingLayers(w5, wires=list(range(n_qubits)))
+            qml.Rot(*w6, wires=3 % n_qubits)
+            qml.RX(w7, wires=4 % n_qubits)
+            return [qml.expval(qml.PauliZ(i)) for i in range(output_dim)]
+
+        layer = KerasLayer(c, w, output_dim=output_dim)
+        x = tf.ones((2, n_qubits))
+
+        layer_out = layer(x)
+        circ_weights = layer.qnode_weights.copy()
+        circ_weights["w4"] = 1.0 * circ_weights["w4"]
+        circ_weights["w6"] = 1.0 * circ_weights["w6"]
```

Comment:
Might be good to leave a comment in the test to indicate why you're doing this

Reply:
Thanks! I followed Josh's suggestion of using `tf.convert_to_tensor()` and also added a comment that this is for slicing.